### ISSUE TITLE
NoContentVerb doesn't allow to provide response headers.

### ISSUE THREAD

**Issue by phadej** [haskell-servant/servant#1267]
This is regression in functionality, yet I don't see an obvious way to improve this.

@alpmestan should I still make the `0.17` with this issue known?

**Comment by alpmestan** [comment]
@phadej I think it's still worth it, it's better than delaying, otherwise the release will just languish if you end up being a bit busy. So good thing that you did, IMO. :-)

**Comment by Taneb** [comment]
I've run into this issue. Could something like adding a variant of `NoContentVerb` which carries with it a list of headers, with `type ServerT (NoContentWithHeaders hs) m = m (Headers hs NoContent)` solve this?

**Comment by danwdart** [comment]
I think that's a good idea. For when I want to have a DeleteNoContent behind an auth for instance.

**Comment by elldritch** [comment]
Yes, I'm running into this as well because I have a `PostNoContent` that sets cookie headers for authentication.

**Comment by torgeirsh** [comment]
We have also run into this, and had to work around it using Verb and a dummy content type that thankfully doesn't cause problems for the HTTP client.
